### PR TITLE
Remove unnecessary TearDowns, doc BaseTest tear down

### LINKED ISSUE #1278: BaseTestWithParam uses destructor to TearDown
```c++
  ~BaseTestWithParam() {
    // Reset scheduler first so that all tasks are done before we kill the StorageManager
    CurrentScheduler::set(nullptr);

#if HYRISE_NUMA_SUPPORT
    // Also make sure that the tasks in the NUMAPlacementManager are not running anymore. We don't restart it here.
    // If you want the NUMAPlacementManager in your test, start it yourself. This is to prevent migrations where we
    // don't expect them
    NUMAPlacementManager::get().pause();
#endif

    PluginManager::reset();
    StorageManager::reset();
    TransactionManager::reset();
    SQLQueryCache<SQLQueryPlan>::get().clear();
    SQLQueryCache<std::shared_ptr<AbstractLQPNode>>::get().clear();
  }
```

imo, this should happen in TearDown() if we want to keep Tests side-effect-free - it is not specified -AFAIK - when the destructor of the Test is run.

### PR DESCRIPTION
**Pull request by mrzzzrm**
Closes #1278

### CODE REVIEW

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -108,7 +108,7 @@ set(
     optimizer/enumerate_ccp_test.cpp
     optimizer/join_graph_builder_test.cpp
     optimizer/join_graph_test.cpp
-    optimizer/lqp_translator_test.cpp
+logical_query_plan/lqp_translator_test.cpp
```
Comment (mrks) [inline src/test/CMakeLists.txt]: You really want me to write a lint script for that, don't you? :D

File: src/test/base_test.hpp [hunk 2]
```diff
@@ -67,6 +67,11 @@ class BaseTestWithParam
 #endif
   }
 
+  /**
+   * Base test uses its destructor over TearDown() to clean up. This way, derived test classes can override TearDown()
```
Comment (mrks) [inline src/test/base_test.hpp]: "instead of"?

File: src/test/expression/lqp_select_expression_test.cpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ using namespace opossum::expression_functional;  // NOLINT
 
 namespace opossum {
 
-class LQPSelectExpressionTest : public ::testing::Test {
```
Comment (mrks) [inline src/test/expression/lqp_select_expression_test.cpp]: On a more serious note: Should we add `::testing::Test` to the linter or do we hope that now that we got rid of all remains, no one will reintroduce it?

Comment (mrzzzrm) [reply src/test/expression/lqp_select_expression_test.cpp]: I think there are still some simple tests (not loading any tables) that still derive from ::testing::Test. Up to you

Comment (mrks) [reply src/test/expression/lqp_select_expression_test.cpp]: ok

**Pull request merged**